suc/update-connection-info.ps1

```diff
@@ -0,0 +1,172 @@
+#>
+
+function Write-LogInfo {
+    Write-Host -NoNewline "INFO: "
+    Write-Host ($args -join " ")
+}
+function Write-LogWarn {
+    Write-Host -NoNewline "WARN: "
+    Write-Host ($args -join " ")
+}
+function Write-LogError {
+    Write-Host -NoNewline "ERROR: "
+    Write-Host ($args -join " ")
+}
+function Write-LogFatal {
+    Write-Host -NoNewline "FATA: "
+    Write-Host ($args -join " ")
+    exit 255
+}
+
+function Write-LogDebug {
+    if (-Not $env:CATTLE_WINS_DEBUG) {
+        Write-Host -NoNewline "DEBUG: "
+        Write-Host ($args -join " ")
+    }
+}
+
+function Test-CaCheckSum() {
+    $caCertsPath = "cacerts"
+    $env:RANCHER_CERT = Join-Path -Path $env:CATTLE_AGENT_CONFIG_DIR -ChildPath "ranchercert"
+    if (-Not $env:CATTLE_CA_CHECKSUM) {
+        return
+    }
+
+    curl.exe --insecure -sfL $env:CATTLE_SERVER/$caCertsPath -o $env:RANCHER_CERT
+    if (-Not(Test-Path -Path $env:RANCHER_CERT)) {
+        Write-Error "The environment variable CATTLE_CA_CHECKSUM is set but there is no CA certificate configured at $( $env:CATTLE_SERVER )/$( $caCertsPath )) "
+        exit 1
+    }
+
+    if ($LASTEXITCODE -ne 0) {
+        Write-Error "Value from $( $env:CATTLE_SERVER )/$( $caCertsPath ) does not look like an x509 certificate, exited with $( $LASTEXITCODE ) "
+        Write-Error "Retrieved cacerts:"
+        Get-Content $env:RANCHER_CERT
+        exit 1
+    }
+    else {
+        Write-LogInfo "Value from $( $env:CATTLE_SERVER )/$( $caCertsPath ) is an x509 certificate"
+    }
+    $env:CATTLE_SERVER_CHECKSUM = (Get-FileHash -Path $env:RANCHER_CERT -Algorithm SHA256).Hash.ToLower()
+    Write-LogInfo "Checking CATTLE_CA_CHECKSUM"
+    if ($env:CATTLE_SERVER_CHECKSUM -ne $env:CATTLE_CA_CHECKSUM) {
+        Remove-Item -Path $env:RANCHER_CERT -Force
+        Write-LogError "Configured cacerts checksum $( $env:CATTLE_SERVER_CHECKSUM ) does not match given -CaCheckSum $( $env:CATTLE_CA_CHECKSUM ) "
+        Write-LogError "Please check if the correct certificate is configured at $( $env:CATTLE_SERVER )/$( $caCertsPath ) ."
+        exit 1
+    }
+    Import-Certificate -FilePath $env:RANCHER_CERT -CertStoreLocation Cert:\LocalMachine\Root | Out-Null
+}
+
+function Test-RancherConnection {
+    $env:RANCHER_SUCCESS = $false
+    $retries = 0
+    while ($retries -lt 6) {
+        $responseCode = $(curl.exe --connect-timeout 60 --max-time 60 -k --write-out "%{http_code}\n" $env:CURL_CAFLAG -sfL "$env:CATTLE_SERVER/healthz")
+        switch ( $responseCode ) {
+            { $_ -in "ok200", 200 } {
+                Write-LogInfo "Successfully tested Rancher connection."
+                $env:RANCHER_SUCCESS = $true
+                $retries = 99
+                break
+            }
+            default {
+                Write-LogError "$responseCode received while testing Rancher connection. Sleeping for 5 seconds and trying again."
+                Start-Sleep -Seconds 5
+                $retries++
+                continue
```

Comment:
The same suggestion to other similar while loop in other functions:
Could we log an addition error message to indicate the test fails because hitting the retries limit (5)?

Reply:
I've added more logging to the various loops, let me know what you think